File: java/src/main/java/io/fivexer/sdk/model/MatchingPolicy.java
package io.fivexer.sdk.model;

/**
 * The live balancer policy applied to bulk matching passes.
 *
 * <p>Populated by Gson via field reflection and exposed read-only; callers receive instances
 * from the client and never construct them.
 */
public class MatchingPolicy {
    private String fairness;  // first-come | best-match | balanced | spread-work
    private Integer maxTasksPerWindow;
    private Long windowMs;

    public String getFairness() { return fairness; }
    public Integer getMaxTasksPerWindow() { return maxTasksPerWindow; }
    public Long getWindowMs() { return windowMs; }
}

File: php/src/Model/MatchingPolicy.php
<?php

declare(strict_types=1);

namespace Fivexer\SDK\Model;

/**
 * The live balancer policy applied to bulk matching passes.
 *
 * Response model: built from a decoded JSON array and read-only thereafter.
 */
final class MatchingPolicy
{
    public function __construct(
        /** first-come | best-match | balanced | spread-work */
        public readonly string $fairness,
        public readonly ?int $maxTasksPerWindow,
        public readonly ?int $windowMs,
    ) {
    }

    /**
     * @param array<string, mixed> $data
     * @internal
     */
    public static function fromArray(array $data): self
    {
        return new self(
            fairness: (string) ($data['fairness'] ?? 'first-come'),
            maxTasksPerWindow: isset($data['maxTasksPerWindow']) ? (int) $data['maxTasksPerWindow'] : null,
            windowMs: isset($data['windowMs']) ? (int) $data['windowMs'] : null,
        );
    }
}
